presto-orc/src/main/java/com/facebook/presto/orc/reader/DecimalStreamReader.java

```diff
@@ -237,13 +240,264 @@ public String toString()
     public void close()
     {
         systemMemoryContext.close();
-        nullVector = null;
-        scaleVector = null;
     }
 
     @Override
     public long getRetainedSizeInBytes()
     {
-        return INSTANCE_SIZE + sizeOf(nullVector) + sizeOf(scaleVector);
+        return INSTANCE_SIZE;
+    }
+
+    @Override
+    public void setFilterAndChannel(Filter filter, int channel, int columnIndex, Type type)
+    {
+        shortDecimal = ((DecimalType) type).isShort();
```

Comment:
Don't we need to validate the `Type` here too?

Reply:
Instead of the repeated `shortDecimal ? 1 : 2` calls everywhere, should we add a field like `numberOfLongs`?

Reply:
@tdcmeehan `ShortDecimalType` and `LongDecimalType` types are not publicly visible.

Reply:
I meant instead of a `ClassCastException` should we do a `checkArgument` that the `type` is actually a `DecimalType`, like the others

Reply:
@tdcmeehan Done.